### DISCUSSION TITLE
DropDown search & GtkExpression

### DISCUSSION THREAD

**Discussion by jgillich** [diamondburned/gotk4#125] [Help] [answered]
I'm trying to use DropDown with search enabled: https://docs.gtk.org/gtk4/method.DropDown.set_enable_search.html

This requires setting an expression. I haven't been able to figure out how to use this, and I read that it's not supported in PyGObject. Does anyone know if this is possible in gotk? Thanks

**Comment by diamondburned** [comment, marked as answer]
gotk4 has the needed constructors for `ConstantExpression`, `PropertyExpression` and `ObjectExpression`. It is only missing a constructor for `ClosureExpression`. It looks like if you're using it with a `StringList` model, you'd want `this->item->string`, which uses `PropertyExpression`s. Anything else would probably require subclassing.

**Reply by diamondburned** [reply]
This looks like a painful API though. I don't know what they're thinking.

**Reply by jgillich** [reply]
Huh, this works:

```go
gtk.NewDropDown(gtk.NewStringList([]string{}), gtk.NewPropertyExpression(gtk.GTypeStringObject, nil, "string"))
```

It does print a few assertion errors, but I'll figure those out later :smile: Thanks for your help
